src/Components/Scenes/Hydration/JobDesigner/DiagramApp/DiagramApp.js

```diff
@@ -0,0 +1,34 @@
+import { DiagramEngine, DiagramModel } from 'storm-react-diagrams';
+
+import { SimplePortFactory } from './SimplePortFactory';
+import { SourceNodeFactory, SourcePortModel } from './SourceNode';
+import { DestNodeFactory, DestPortModel } from './DestNode';
+import { TransNodeFactory, TransPortModel } from './TransNode';
+
+export default class DiagramApp {
+  constructor() {
+    this.diagramEngine = new DiagramEngine();
+    this.diagramEngine.installDefaultFactories();
```

Comment:
do we know if we need the default factories here as well? just from the name it implies that we are installing additional factories along with the custom ones we made. Isn't a super big deal, just curious.

Reply:
Good q! I'll look into this.

Reply:
So what I discovered is...although we have custom nodes and ports, we are still using the default link model for now, so the default factories are a good catch-all to make sure that's all loaded in correctly.

Reply:
makes sense, thanks for looking into it further!